src/hoodie/request.js

```diff
@@ -64,46 +66,48 @@ exports.request = function(hoodie, type, url, options) {
     url = (hoodie.baseUrl || '') + API_PATH + url;
   }
 
-  // if url is cross domain, set CORS headers
-  if (/^http/.test(url)) {
-    defaults.xhrFields = {
-      withCredentials: true
-    };
-    defaults.crossDomain = true;
-  }
-
   defaults.url = url;
+  
+  //passing `data` as an option is throughout the code
+  if(options.data){
+    options.body = options.data;
+    delete options.data;
+  }
 
-  // we are piping the result of the request to return a nicer
-  // error if the request cannot reach the server at all.
-  // We can't return the promise of ajax directly because of
-  // the piping, as for whatever reason the returned promise
-  // does not have the `abort` method any more, maybe others
-  // as well. See also http://bugs.jquery.com/ticket/14104
-  jQueryPromise = global.jQuery.ajax(extend(defaults, options))
-    .done(requestDefer.resolve)
-    .fail(requestDefer.reject);
-  var pipedPromise = requestPromise.catch(exports.handleRequestError.bind(null, hoodie));
-  pipedPromise.abort = jQueryPromise.abort;
-
-  return pipedPromise;
+  var ajaxRequest = ajax(extend(defaults, options), function(err, response, data){
+    if(err){
+      return requestDefer.reject(err);
+    }
+    requestDefer.resolve(response, data);
+  });
+  var promise = requestPromise.catch(exports.handleRequestError.bind(null, hoodie));
+  
+  promise.abort = ajaxRequest.abort;
```

Comment:
It seems to me like an anti-pattern to attach `abort` to the promise, but I didn't want to take on too much in this first contribution, so I left it be. 
The only place I see it being used is in `account/helpers.js:withPreviousRequestsAborted` - maybe instead of returning the abort method it should be added to a global queue, or maybe this method should return an object:
```{
promise: promise,
abort: ajaxRequest.abort
}
```
Thoughts?